File: python/diccionarios/clasificar_contactos.py
agenda = {
    "Juan": {"telefono": "3001234567", "correo": "juan@gmail.com", "categoria": "Familia"},
    "Maria": {"telefono": "3019876543", "correo": "maria@gmail.com", "categoria": "Trabajo"},
    "Pedro": {"telefono": "3025554444", "correo": "pedro@gmail.com", "categoria": "Amigos"},
    "Ana": {"telefono": "3008887777", "correo": "ana@gmail.com", "categoria": "Familia"}
}
print("Contactos clasificados por categoría:\n")
for nombre, datos in agenda.items():
    print("Nombre:", nombre)
    print("Teléfono:", datos["telefono"])
    print("Correo:", datos["correo"])
    print("Categoría:", datos["categoria"])
    print("----------------------")

File: visual_basic/colecciones_complejas/clasificar_contactos/clasificar_contactos.vb
Imports System
Imports System.Collections.Generic
Module clasificar_contactos
    Class Contacto
        Public Nombre As String
        Public Telefono As String
        Public Correo As String
        Public Categoria As String
    End Class
    Sub Main(args As String())
        Dim agenda As New Dictionary(Of String, List(Of Contacto))
        AgregarContacto(agenda, New Contacto With {.Nombre = "Juan", .Telefono = "123", .Correo = "juan@mail.com", .Categoria = "Familia"})
        AgregarContacto(agenda, New Contacto With {.Nombre = "Maria", .Telefono = "456", .Correo = "maria@mail.com", .Categoria = "Trabajo"})
        AgregarContacto(agenda, New Contacto With {.Nombre = "Carlos", .Telefono = "789", .Correo = "carlos@mail.com", .Categoria = "Amigos"})
        AgregarContacto(agenda, New Contacto With {.Nombre = "Ana", .Telefono = "321", .Correo = "ana@mail.com", .Categoria = "Familia"})
        MostrarContactosClasificados(agenda)
        Console.ReadKey()
    End Sub
    Sub AgregarContacto(agenda As Dictionary(Of String, List(Of Contacto)), contacto As Contacto)
        If Not agenda.ContainsKey(contacto.Categoria) Then
            agenda(contacto.Categoria) = New List(Of Contacto)
        End If
        agenda(contacto.Categoria).Add(contacto)
    End Sub
    Sub MostrarContactosClasificados(agenda As Dictionary(Of String, List(Of Contacto)))
        Console.WriteLine("📒 CONTACTOS CLASIFICADOS")
        Console.WriteLine("===================================")
        For Each categoria In agenda.Keys
            Console.WriteLine("📂 Categoría: " & categoria)
            Console.WriteLine("---------------------------")
            For Each contacto In agenda(categoria)
                Console.WriteLine("Nombre: " & contacto.Nombre)
                Console.WriteLine("Teléfono: " & contacto.Telefono)
                Console.WriteLine("Correo: " & contacto.Correo)
                Console.WriteLine("---------------------------")
            Next
            Console.WriteLine()
        Next
    End Sub
End Module